shell/opslevel/upload.sh

```diff
@@ -21,7 +21,26 @@ source "$DIR/../lib/opslevel.sh"
 
 # Download Artifacts JSON once
 ARTIFACTS_JSON_FILE=/tmp/artifacts.json
-download_artifacts_json "${ARTIFACTS_JSON_FILE}"
+
+# have a loop that runs every 2 seconds and will terminate after X seconds or if file doesnt contain message
+# Set interval (duration) in seconds.
+SECONDS=20
+# Calculate end timeout.
+ENDTIMEOUT=$(($(date +%s) + SECONDS))
+
+while [ "$(date +%s)" -lt $ENDTIMEOUT ]; do
+  sleep 2
```

Comment:
Putting the sleep after line 40 will make this 2 seconds faster.

Reply:
You edited the wrong `sleep`. :)

You also still want this - but you want to move it to only happen after failure (line 40 in this file).